### ISSUE TITLE
Inconsistent scanning of google-drive-ocamlfuse folders

### ISSUE THREAD

**Issue by Kabouik** [airsonic/airsonic#772]
## Problem description

When using a [google-drive-ocamlfuse](https://github.com/astrada/google-drive-ocamlfuse) folder as Music folder for the library, the files therein may or may not be detected by Airsonic. I thought it was perhaps because the access to the files would be too slow, but turns out [some people have success streaming video content using this method](https://www.effinspiders.com/setting-up-plex-media-server-on-an-ubuntu-linux-vps-with-google-drive/), which probably requires higher bandwidth.

### Steps to reproduce

1. Install `google-drive-ocamlfuse` as airsonic, authenticate (see their wiki)
2. `mkdir /path/to/mountpoint` as airsonic user
3. Add album folders to the corresponding Google Drive
3. Check ownership of the folder, mount the distant Google Drive to it using `google-drive-ocamlfuse /path/to/mountpoint`
4. Check that files are showing up in `ls -la /path/to/mountpoint``
5. `java -jar /var/airsonic/airsonic.war` as airsonic user, or use a systemd service.
6. Add a music folder for `/path/to/mountpoint`
7. Hit the link to manually scan.
8. Pour yourself a coffee and come back later to see files have been found. 
9. `cat /var/airsonic/airsonic.log`

I *did* have success for some albums folders, but Airsonic most often misses them when scanning, and numerous tries won't change a thing. At the moment I have only this Google Drive as a music folder in Airsonic, and some albums are detected, others not, even after a million attempts to scan and several hours. I can't see any clear pattern when looking at the files themselves (but from my testing, file format and accented characters can be ruled out). 

## System information

 * **Airsonic version**: 10.1.2
 * **Operating system**: Debian Jessie, VPS
 * **Proxy server**: nginx, but most likely not related
 * **Client**: Vivaldi, Firefox, Ultrasonic application

## Additional notes

*En of the /var/airsonic/airsonic.log:* 

```
2018-08-01 17:51:00.022  INFO --- o.a.p.c.StreamController                 : Got HTTP range: 13729183-
2018-08-01 17:51:00.041  INFO --- o.a.p.io.PlayQueueInputStream            : admin listening to "Zanaka/03. Jain - Mr Johnson.flac"
2018-08-01 17:55:19.566  INFO --- o.a.p.s.MediaScannerService              : Starting to scan media library.
2018-08-01 17:55:20.100  INFO --- o.a.p.s.MediaScannerService              : Scanned media library with 59 entries.
2018-08-01 17:55:20.100  INFO --- o.a.p.s.MediaScannerService              : Marking non-present files.
2018-08-01 17:55:20.102  INFO --- o.a.p.s.MediaScannerService              : Marking non-present artists.
2018-08-01 17:55:20.103  INFO --- o.a.p.s.MediaScannerService              : Marking non-present albums.
2018-08-01 17:55:20.107  INFO --- o.a.p.s.MediaScannerService              : Completed media library scan.
2018-08-01 17:55:20.479  INFO --- o.a.p.service.PlaylistService            : Starting playlist import.
2018-08-01 17:55:20.479  INFO --- o.a.p.service.PlaylistService            : Completed playlist import.
2018-08-01 17:55:51.169  INFO --- o.a.p.s.MediaScannerService              : Starting to scan media library.
2018-08-01 17:55:51.775  INFO --- o.a.p.s.MediaScannerService              : Scanned media library with 59 entries.
2018-08-01 17:55:51.776  INFO --- o.a.p.s.MediaScannerService              : Marking non-present files.
2018-08-01 17:55:51.777  INFO --- o.a.p.s.MediaScannerService              : Marking non-present artists.
2018-08-01 17:55:51.792  INFO --- o.a.p.s.MediaScannerService              : Marking non-present albums.
2018-08-01 17:55:51.806  INFO --- o.a.p.s.MediaScannerService              : Completed media library scan.
2018-08-01 17:55:52.052  INFO --- o.a.p.service.PlaylistService            : Starting playlist import.
2018-08-01 17:55:52.052  INFO --- o.a.p.service.PlaylistService            : Completed playlist import.
```
The logs shows two attempts to scan the music folder, but airsonic finishes the scan less than one second later, and in fact it is missing 50 .mp3 files in that case.

**Comment by Kabouik** [comment]
I have created an [issue](https://github.com/astrada/google-drive-ocamlfuse/issues/470) on the [google-drive-ocamlfuse](https://github.com/astrada/google-drive-ocamlfuse) Github project page too. 

However, please note that despite the issue described there (Google Drive changes not reflecting in the mount point), the scanning problem I am describing above for Airsonic is for files that **do** show up in the mountpoint. They are visible from the terminal, and I can read them (I tried with `sha256sum`), but Airsonic won't scan them.

Now could it be that files that don't show up in terminal and files that do but are not seen by Airsonic are caused by the same thing? No idea, but I created an issue on both Github projects so we can see how interrelated they are.

**Comment by Kabouik** [comment]
There are now 2880 audio or cover.jpg files visible from the terminal in my `/path/to/mountpoint` (totalling 15 GB of content instead of 1 GB when I opened this issue), but Airsonic only imported 50 tracks so far, belonging to 4 folders. The drive is made of folders in the the Artist/Album/Tracks structure. Still no apparent pattern as to why these 4 folders were initially detected and not all those I added later, but it is noteworthy that they were detected early after I installed Airsonic, whereas all subsequent scans found nothing. 

Interestingly, I added an album into one of the few artist folders that were initially detected, and this album was the only one being scanned by Airsonic when the content of the music folder went from 1 GB to 15 GB.

**Comment by jooola** [comment]
@Kabouik Could you send a screenprint of your media folder settings ?

Alos I would try to wipe the $airsonic_home and restart fresh without touching your audio files.

**Comment by Kabouik** [comment]
![screenshot from 2018-08-02 18 11 43](https://user-images.githubusercontent.com/7107523/43596438-995ebd90-967f-11e8-8b3f-89186c2ce9d4.png)

I can try deleting /var/airsonic and restarting from scratch, will let you know how it goes.

And yeah, I should definitely update to strech, but I'm afraid a dist-upgrade would do more harm than good, and once I try, there is no way back. :<

**Comment by Kabouik** [comment]
Now that was interesting:

![](https://reho.st/0x0.st/sWgj.png)

In the logs I just saw errors about some tags that could not be read properly, but I think the files imported properly. I haven't checked folders one by one yet, but it seems that some stuff is missing (I expected more tracks, and more than 14.5 GB).

Maybe the issue is related to #732 too.

**Comment by Kabouik** [comment]
I added a new artist to the media folder, it was immediately visible in terminal, but Airsonci won't scan it. It really looks like #732.

**Comment by titooo7** [comment]
I'm also facing this issue. 
My music is stored in a google-ocamlfuse  mounted folder. The owner of the folder mounted and the user running airsonic are the same ones, so I don't think it's a permissions issue, but I don't know what it could be...

**Comment by pjsurk** [comment]
It is because Airsonic looks at the dates of the folders. If they are not changed it skips the folder. Google drive however does not update the date of the parent folder if you put a new file/folder inside. 

If you rename the parent folder and rename back it will give a new date and the files will be scanned correctly.

**Comment by muff1nman** [comment]
See #607

**Comment by tesshucom** [comment]
Assuming that the option to disable time stamps during the scan is enabled will not cause any issues with scanning in Google Drive?
If you have confirmed the operation, please give us your feedback.

**Comment by Kabouik** [comment]
Sorry I missed the new post in this issue earlier. Could you please explain how to enable this option to ignore time stamps during scan?

**Comment by tesshucom** [comment]
> See #607

Is it possible to trace the process from here?

**Comment by Kabouik** [comment]
If I understood correctly, I need do edit /var/airsonic/airsonic.properties and add `IgnoreFileTimestamps=true`.

However it seems my google-drive-ocamlfuse Google authorization has been revoked (in the end I could not use this solution due to the present issue). Re-enabling it implies some more work on a headless server because I cannot use a browser to authenticate, and the option suggested by omcalfuse developers for headless machines restricts access to a subdirectory and to files uploaded using ocamlfuse, which is not what I currently have on the Google Drive. I may need to start from scratch but I kind of forgot how google-drive-ocamlfuse works, so I will have to read a bit through the documentation first.

**Comment by tesshucom** [comment]
Certainly, I don't understand the current problems and end points.
It was pointed out that there was a problem with scanning when used in combination with a specific product, and there was a problem with time determination, and a workaround was suggested.

Please let me know if you have any clues.

**Comment by Kabouik** [comment]
Absolutely. I suspect your workaround would solve the issue I was having. However, I will need some time to read through the documentation of the other product to make it work again and test the workaround, since ocamlfuse is no longer in a working state on my machine and I forgot the details over time since I couldn't use it as I wanted. No problem for me if you prefer to close the issue in the mean time, I don't want to hold you back here.

**Comment by tesshucom** [comment]
We understand.
It's an old topic, so it's unclear if it should be urgent now. 

I wasn't the party to the problem, and it seemed that it was left objectively resolved.
We will close the problem once, so please report any new problems.
I may be able to help.

**Comment by pjsurk** [comment]
However this setting is not really helpful for Gdrive. Everytime it will
need to full scan and read the complete library folders again which will
take an extreme long time to process.

A better method would be if Airsonic scanned the timestamp of all folders
instead of just the root folders and then imported the new ones.

Op ma 25 mei 2020 om 15:07 schreef tesshu.com <notifications@github.com>:
